### PR TITLE
add filter for country

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**
Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 1]
```diff
@@ -151,7 +162,8 @@ private static String sanitizeParam(String param) {
                 .filter(branch -> matches(filtersContingencyList.getEquipmentID(), branch.getId()) || branch.getOptionalName().isPresent() && matches(filtersContingencyList.getEquipmentName(), branch.getOptionalName().get()))
                 .filter(branch -> filtersContingencyList.getNominalVoltage() == -1 || filterByVoltage(branch.getTerminal1().getVoltageLevel().getNominalV(), filtersContingencyList.getNominalVoltage(), filtersContingencyList.getNominalVoltageOperator())
                         || filterByVoltage(branch.getTerminal2().getVoltageLevel().getNominalV(), filtersContingencyList.getNominalVoltage(), filtersContingencyList.getNominalVoltageOperator()))
-                .map(branch -> new Contingency(branch.getId(), Collections.singletonList(new BranchContingency(branch.getId()))))
+                .filter(injection -> countryFilter(injection, filtersContingencyList))
```
Comment (ChamseddineBhd) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:165]: rename injection -> branch

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java [hunk 2]
```diff
@@ -34,12 +39,15 @@
 
     private String nominalVoltageOperator;
 
+    private List<String> countries;
```
Comment (ChamseddineBhd) [inline src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java:42]: Set<String>

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java [hunk 3]
```diff
@@ -34,12 +39,15 @@
...
+
     public FiltersContingencyListEntity(String name, FiltersContingencyListAttributes filtersContingencyListAttributes) {
         this.name = name;
         this.equipmentId = filtersContingencyListAttributes.getEquipmentID();
         this.equipmentName = filtersContingencyListAttributes.getEquipmentName();
         this.equipmentType = filtersContingencyListAttributes.getEquipmentType();
         this.nominalVoltage = filtersContingencyListAttributes.getNominalVoltage();
         this.nominalVoltageOperator = filtersContingencyListAttributes.getNominalVoltageOperator();
+        this.countries = emptyIfNull(filtersContingencyListAttributes.getCountries());
```
Comment (ChamseddineBhd) [inline src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java:51]: new Haset(...)
Save data as set to not have duplicated stored elements

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 4]
```diff
@@ -65,7 +65,7 @@ private static ScriptContingencyList fromScriptContingencyListEntity(ScriptConti
 
     private static FiltersContingencyList fromFilterContingencyListEntity(FiltersContingencyListEntity entity) {
         return new FiltersContingencyList(entity.getName(), entity.getEquipmentId(), entity.getEquipmentName(),
-                entity.getEquipmentType(), entity.getNominalVoltage(), entity.getNominalVoltageOperator());
+                entity.getEquipmentType(), entity.getNominalVoltage(), entity.getNominalVoltageOperator(), new HashSet<>(entity.getCountries()));
```
Comment (ChamseddineBhd) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:68]: set countries field as set then entity. new HashSet<>(entity.getCountries()) -> getCountries()

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java (hunk 1 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/ContingencyListService.java]: done

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java (hunk 2 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java]: done

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java (hunk 4 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/ContingencyListService.java]: done

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java (hunk 3 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java]: done

**Review by ChamseddineBhd — APPROVED** [review]

**Pull request merged**